services/worker-manager/src/api.js

```diff
@@ -586,11 +633,29 @@ builder.declare({
   return res.reply(result);
 });
 
-let cleanPayload = payload => {
+const cleanPayload = payload => {
   payload = '(OMITTED)';
   return payload;
 };
 
+const createTemporaryCredentials = ({worker, expires, permanentCredentials}) => {
+  return taskcluster.createTemporaryCredentials({
+    clientId: `worker/${worker.providerId}/${worker.workerPoolId}/${worker.workerGroup}/${worker.workerId}`,
+    scopes: [
+      `assume:worker-type:${worker.workerPoolId}`, // deprecated role
+      `assume:worker-pool:${worker.workerPoolId}`,
+      `assume:worker-id:${worker.workerGroup}/${worker.workerId}`,
+      `queue:worker-id:${worker.workerGroup}/${worker.workerId}`,
+      `secrets:get:worker-type:${worker.workerPoolId}`, // deprecated secret name
+      `secrets:get:worker-pool:${worker.workerPoolId}`,
+      `queue:claim-work:${worker.workerPoolId}`,
```

Comment:
I was thinking we would land all of this to get the "misconfigured things are terminated" stuff and then go back and add reregistering in a followup since that bit is less important.

Reply:
Sounds good